sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgExpressionService.cs

```diff
@@ -19,31 +19,66 @@
  */
 
 using System.Collections.Generic;
+using System.Linq;
 using Microsoft.CodeAnalysis;
 using SonarAnalyzer.Helpers;
-using SonarAnalyzer.Protobuf.Ucfg;
 
 namespace SonarAnalyzer.ControlFlowGraph.CSharp
 {
     internal class UcfgExpressionService
     {
-        private readonly Dictionary<SyntaxNode, Expression> syntaxNodeToUcfgExpressionCache
-            = new Dictionary<SyntaxNode, Expression>();
+        private static readonly ISet<KnownType> PrimitiveTypes = new[] { KnownType.System_Boolean }
+             .Union(KnownType.IntegralNumbers)
+             .Union(KnownType.NonIntegralNumbers)
+             .ToHashSet();
 
-        public void RegisterAsThis(SyntaxNode syntaxNode) =>
-            syntaxNodeToUcfgExpressionCache[syntaxNode] = UcfgExpression.This;
+        private readonly Dictionary<SyntaxNode, UcfgExpression> cache
+            = new Dictionary<SyntaxNode, UcfgExpression>();
 
-        public void RegisterAsVariable(SyntaxNode syntaxNode, string variableName) =>
-            syntaxNodeToUcfgExpressionCache[syntaxNode] = UcfgExpression.FromVariable(variableName);
+        private int numberedVariableCounter;
 
-        public void RegisterAsConstant(SyntaxNode syntaxNode) =>
-            syntaxNodeToUcfgExpressionCache[syntaxNode] = UcfgExpression.Constant;
+        public void Associate(SyntaxNode syntaxNode, UcfgExpression expression) =>
+            cache[syntaxNode.RemoveParentheses()] = expression;
```

Comment:
Are there any test cases that cover expressions that differ only by parentheses?